client.go

```diff
@@ -50,10 +50,14 @@ func HttpService(rt http.RoundTripper) Service {
 		// When the calling context is cancelled, close the response body
 		// This protects callers that forget to call Close(), or those which proxy responses upstream
 		if httpRsp != nil && httpRsp.Body != nil {
-			body := httpRsp.Body
+			dr := newDoneReader(httpRsp.Body)
+			httpRsp.Body = dr
 			go func() {
-				<-req.Done()
-				body.Close()
+				select {
+				case <-dr.done:
```

Comment:
Do we _also_ need to close the body in the first case too (whilst ensuring that we only close it once)? If we incur an `io.EOF` whilst reading, then we will close the done channel but the body won't be closed.